File: app/src/main/java/com/leodeleon/popmovies/data/local/MovieDB.java
package com.leodeleon.popmovies.data.local;

import com.leodeleon.popmovies.model.Movie;
import io.reactivex.Completable;
import io.reactivex.Single;
import java.util.List;

public class MovieDB {

  private MovieDao movieDao;

  public MovieDB(PopMoviesDB popMoviesDB) {
    movieDao = popMoviesDB.movieDao();
  }

  public Completable saveMovie(Movie movie) {
   return Completable.create(completableEmitter -> {
     movieDao.insert(movie);
     completableEmitter.onComplete();
   });
  }

  public Completable deleteMovie(Movie movie) {
    return Completable.create(completableEmitter -> {
      movieDao.delete(movie);
      completableEmitter.onComplete();
    });
  }

  public Single<List<Movie>> getFavMovies() {
    return movieDao.getFavoriteMovies().onBackpressureBuffer().firstOrError();
  }

}

File: app/src/main/java/com/leodeleon/popmovies/data/local/MovieDB.kt
package com.leodeleon.popmovies.data.local

import com.leodeleon.popmovies.model.Movie
import io.reactivex.Completable
import io.reactivex.Single

class MovieDB(popMoviesDB: PopMoviesDB) {

  private var movieDao: MovieDao = popMoviesDB.movieDao()

  fun saveMovie(movie: Movie): Completable {
    return Completable.create { completableEmitter ->
      movieDao.insert(movie)
      completableEmitter.onComplete()
    }
  }

  fun deleteMovie(movie: Movie): Completable {
    return Completable.create { completableEmitter ->
      movieDao.delete(movie)
      completableEmitter.onComplete()
    }
  }

  fun getFavoriteMovies(): Single<List<Movie>> = movieDao.getFavoriteMovies().onBackpressureBuffer().firstOrError()

}
